Python/compile.c

```diff
@@ -184,19 +184,40 @@ static struct jump_target_label_ NO_LABEL = {-1};
         return 0; \
     }
 
+typedef enum {
+    UNUSED_ARG,       /* No oparg */
+    EXPLICIT_ARG,     /* An int value */
+    CONST_ARG,        /* Index into co_consts */
+    JUMP_TARGET_ARG,  /* ID of a jump target block */
+} oparg_type;
+
+typedef struct {
+    oparg_type type;
+    int value;
+} oparg_t;
+
+#define OPARG_VALUE(O) ((O).value)
```

Comment:
If you make `OPARG_VALUE` a static inline function, you can put the `assert(arg_type_is_compatible(instr));` here and clean up the uses of the function.

Reply:
I guess if we're sticking with one oparg we can do that (my intention was not to pass in the whole instr).

Reply:
On second thought I'm not sure this PR is worth merging. It was supposed to be a step towards the register machine, but while we stay with a stack machine I think it's making the code look over-engineered and complicated for little benefit.  So I suggest we park this and revisit when we want to work on the register machine again.